packages/model-viewer/src/three-components/gltf-instance/correlated-scene-graph.ts

```diff
@@ -277,10 +283,14 @@ export class CorrelatedSceneGraph {
          gltfMaterialIndex: number) => {
           updatedMaterials.add(gltfMaterialIndex);
           const gltfElement = this.gltf.materials![gltfMaterialIndex];
-          let threeObjects = this.gltfElementMap.get(gltfElement);
+          if (this[$gltfVariantMap] === null) {
+            this[$gltfVariantMap] = new Map<GLTFElement, ThreeObjectSet>();
```

Comment:
Seems like `$gltfVariantMap` is duplicating `$gltfElementMap` in  much  the same way as was happing with the  array and map on `model`. Can we simplify this in the same way? Is there any reason we need both?

Reply:
This prevents bad correlations, I'm using the new map to prevent variant objects with the same key from being correlated together. Actually looking at this now I think I can see more clearly where the bug was coming from, I think I might be able to get rid of it.